modules/core/src/main/java/org/apache/ignite/lang/ErrorGroups.java

```diff
@@ -117,92 +117,35 @@ public static class Sql {
-        public static final int PRIMARY_KEYS_MULTIPLE_ERR = SQL_ERR_GROUP.registerErrorCode(6);
-
         /** Schema not found. */
-        public static final int SCHEMA_NOT_FOUND_ERR = SQL_ERR_GROUP.registerErrorCode(7);
-
-        /** Storage engine not valid. */
-        public static final int STORAGE_ENGINE_NOT_VALID_ERR = SQL_ERR_GROUP.registerErrorCode(8);
+        public static final int SCHEMA_NOT_FOUND_ERR = SQL_ERR_GROUP.registerErrorCode(5);
 
         /** Cursor is already closed error. */
-        public static final int CURSOR_CLOSED_ERR = SQL_ERR_GROUP.registerErrorCode(9);
-
-        /** Constraint violation: some keys can't be inserted because they violate unique constraint PK. */
-        public static final int DUPLICATE_KEYS_ERR = SQL_ERR_GROUP.registerErrorCode(10);
-
-        /** Constraint violation: deleting a column that belongs to the index. */
-        public static final int DROP_IDX_COLUMN_CONSTRAINT_ERR = SQL_ERR_GROUP.registerErrorCode(11);
-
-        /**  Too many grouping expressions. */
-        public static final int TOO_MANY_GROUPING_EXPRESSIONS_ERR = SQL_ERR_GROUP.registerErrorCode(12);
-
-        /** Unsupported sql operation. */
-        public static final int UNSUPPORTED_SQL_OPERATION_KIND_ERR = SQL_ERR_GROUP.registerErrorCode(13);
-
-        /** Unsupported DDL operation. */
-        public static final int UNSUPPORTED_DDL_OPERATION_ERR = SQL_ERR_GROUP.registerErrorCode(14);
-
-        /** Query validation error. */
-        public static final int QUERY_VALIDATION_ERR = SQL_ERR_GROUP.registerErrorCode(15);
+        public static final int CURSOR_CLOSED_ERR = SQL_ERR_GROUP.registerErrorCode(6);
 
-        /** Object is not found in schema. */
-        public static final int OBJECT_NOT_FOUND_ERR = SQL_ERR_GROUP.registerErrorCode(16);
+        /** Parsing error. */
+        public static final int PARSE_ERR = SQL_ERR_GROUP.registerErrorCode(7);
 
-        /** Object already exists in schema. */
-        public static final int OBJECT_ALREADY_EXISTS_ERR = SQL_ERR_GROUP.registerErrorCode(17);
+        /** Validation error. */
+        public static final int VALIDATION_ERR = SQL_ERR_GROUP.registerErrorCode(8);
 
-        /** Query mapping error. */
-        public static final int QUERY_MAPPING_ERR = SQL_ERR_GROUP.registerErrorCode(19);
+        /** Constraint violation error. */
+        public static final int CONSTRAINT_VIOLATION_ERR = SQL_ERR_GROUP.registerErrorCode(9);
 
-        /** DDL execution error. */
-        public static final int DDL_EXEC_ERR = SQL_ERR_GROUP.registerErrorCode(20);
+        /** Internal SQL engine error. */
+        public static final int INTERNAL_EXECUTION_ERR = SQL_ERR_GROUP.registerErrorCode(10);
 
-        /** DML result error. */
-        public static final int INVALID_DML_RESULT_ERR = SQL_ERR_GROUP.registerErrorCode(21);
+        /** Runtime SQL engine error. */
+        public static final int RUNTIME_EXECUTION_ERR = SQL_ERR_GROUP.registerErrorCode(11);
```

Comment:
We can combine all error to internal errors but if you look more closely on where these code they are used for different situations. I do not have strong opinion on this.

Reply:
I looked at a places where INTERNAL errors are thrown. It looks like unrecoverable internal errors.
I mean that there is no way to fix the issue at runtime and possibly grid should be stopped.

On the other hand, RUNTIME exception allow user to fix query string, create missed entities or just retry because of new topology established, without grid maintenance.

Reply:
Renamed to `RUNTIME_EXECUTION_ERR` to `EXECUTION_ERR`.  Replaced `INTERNAL_EXECUTION_ERR`s with `INTERNAL_ERR`s